pkg/proxy/audit/audit.go

```diff
@@ -0,0 +1,71 @@
+// Copyright Jetstack Ltd. See LICENSE for details.
+package audit
+
+import (
+	"fmt"
+	"net/http"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	genericapifilters "k8s.io/apiserver/pkg/endpoints/filters"
+	"k8s.io/apiserver/pkg/server"
+	genericfilters "k8s.io/apiserver/pkg/server/filters"
+
+	"github.com/jetstack/kube-oidc-proxy/cmd/app/options"
+)
+
+type Audit struct {
+	options      *options.AuditOptions
+	serverConfig *server.CompletedConfig
+}
+
+func New(options *options.AuditOptions, externalAddress string, secureServingInfo *server.SecureServingInfo) (*Audit, error) {
```

Comment:
nit: `options` here shadows the package named `options` which makes this file harder to read/follow (took me some extra passes over) - maybe rename this arg to `opts`?

Reply:
:+1: